proto/def.bzl

```diff
@@ -90,6 +94,8 @@ def _proto_library_to_source(_go, attr, source, merge):
             merge(source, compiler[GoSource])
 
 def _go_proto_library_impl(ctx):
+    check_importpaths(ctx)
```

Comment:
This seems new?

Reply:
Yeah so one of the things bothering me is that there are some differences between the library (base, proto, tool) and binary/test attributes/codepaths, but they both funnel into `go_context`. Then we end up having a bunch of `getattr`/`hasattr` and fallbacks to handle both of these. I wonder if it would be cleaner to have a dedicated lib_context and bin_context that have a shared shape?

Absent that, this change hoists `check_importpath` calls to the library entrypoints, because these attrs don't exist on binaries, and this allows simplifying the check function to avoid `hasattr` which speeds it up.

Without:
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/4d34ada1-da37-484b-90bd-82a941144bf2">

With:
<img width="1725" alt="image" src="https://github.com/user-attachments/assets/1de8eff2-9d3a-4712-b05d-1e341d6827b6">